modules/tinymce/src/core/main/ts/caret/LineUtils.ts

```diff
@@ -22,8 +22,8 @@ export interface CaretInfo {
 
 const isContentEditableFalse = NodeType.isContentEditableFalse;
 const findNode = CaretUtils.findNode;
-const distanceToRectLeft = (clientRect: NodeClientRect, clientX: number) => Math.abs(clientRect.left - clientX);
-const distanceToRectRight = (clientRect: NodeClientRect, clientX: number) => Math.abs(clientRect.right - clientX);
+const distanceToRectLeft = (clientRect: ClientRect, clientX: number) => Math.abs(clientRect.left - clientX);
+const distanceToRectRight = (clientRect: ClientRect, clientX: number) => Math.abs(clientRect.right - clientX);
```

Comment:
Something about this (i.e. the whole file) is bothering me, but I think it's because NodeClientRect extends ClientRect and therefore TS is just figuring things out without us explicitly typing things as I'd expect...

Reply:
Yeah, that's why it works, since `ClientRect` is the base type, so as these functions don't need the subtypes specifically, then we don't need to pass it in as such. There does however seem to be something wrong with this file, as noted by my TODO below.